be/src/exec/pipeline/adaptive/collect_stats_context.cpp

```diff
@@ -0,0 +1,295 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "exec/pipeline/adaptive/collect_stats_context.h"
+
+#include <utility>
+
+#include "column/chunk.h"
+#include "common/statusor.h"
+#include "exec/pipeline/adaptive/adaptive_dop_param.h"
+#include "exec/pipeline/adaptive/utils.h"
+
+namespace starrocks::pipeline {
+
+/// BlockState.
+std::string BlockState::name() const {
+    return "Block";
+}
+
+bool BlockState::need_input(int32_t driver_seq) const {
+    return true;
+}
+
+Status BlockState::push_chunk(int32_t driver_seq, ChunkPtr chunk) {
+    size_t num_chunk_rows = chunk->num_rows();
+    _ctx->_buffer_chunk_queue(driver_seq).emplace(std::move(chunk));
+    size_t prev_num_rows = _num_rows.fetch_add(num_chunk_rows);
+
```

Comment:
here is very important point, please add some comment

Reply:
Fine.